ui/src/components/Header.tsx

```diff
@@ -78,86 +81,79 @@ export default function DashboardNavbar({
-                          {userEmail && (
-                            <div className="border-b border-b-slate-800/20 dark:border-slate-300/20">
-                              <Menu.Item>
-                                <div className="p-2">
-                                  <div className="font-display text-xs">Logged in as</div>
-                                  <div className="w-[200px] truncate font-display text-sm font-semibold">
-                                    {userEmail}
-                                  </div>
-                                </div>
-                              </Menu.Item>
-                            </div>
-                          )}
-                          <div>
-                            <Menu.Item>
-                              <div onClick={onLogout}>
-                                <button className="block w-full">
-                                  <div className="flex items-center gap-x-2 rounded-md px-2 py-1.5 text-sm transition-colors hover:bg-slate-100 dark:hover:bg-slate-700">
-                                    <ArrowLeftEndOnRectangleIcon className="h-4 w-4" />
-                                    <div className="font-display">Log out</div>
+                  </>
+                )}
+                {isLoggedIn ? (
+                  <>
+                    <hr className="h-[20px] w-[1px] border-none bg-slate-800/20 dark:bg-slate-300/20" />
+                    <div className="relative inline-block text-left">
+                      <Menu>
+                        <MenuButton>
+                          <Button className="flex items-center gap-x-3 rounded-md border bg-white dark:border-slate-600 dark:bg-slate-800 dark:text-white border-slate-800/20 px-2 py-1.5">
+                            {picture
+                              ? (
+                                <img
+                                  src={picture}
+                                  alt="Avatar"
+                                  className="size-6 rounded-full border-2 border-transparent transition-colors group-hover:border-blue-700"
+                                />
+                              )
+                              : (
+                                <span className="max-w-[200px] text-sm/6 font-display font-semibold truncate">
+                                  {userEmail}
+                                </span>
+                              )
+                            }
+                            <ChevronDownIcon className="size-4 shrink-0 text-slate-900 dark:text-white" />
+                          </Button>
+                        </MenuButton>
+                        <MenuItems
+                          transition
+                          anchor="bottom end"
+                          className="right-0 mt-1 w-56 origin-top-right data-closed:opacity-0 focus:outline-hidden">
+                          <MenuItem>
+                            <Card className="overflow-hidden">
```

Comment:
Big change here is putting the `Card` inside the `MenuItem` (of which there is now only one) because trying to wrap the `Menu.Items` in the`Card` completely breaks this version of headlessui.